SweetsFactory/src/app/shared/account/account.component.ts

```diff
@@ -1,35 +1,42 @@
-import { Component, Input, OnInit } from '@angular/core';
+import { Component, Input, OnInit, Renderer } from '@angular/core';
 import { UserService } from '../services/user.service';
 import { User } from '../model/user';
+import { Router } from '@angular/router';
 
 @Component({
     selector: 'account',
     templateUrl: './account.component.html',
-    styles: [`
-    .button {
-        border: none;
-    }`]
+    styleUrls: ['./account.component.css']
 })
 
 export class AccountComponent implements OnInit {
     accountName: string = "Account";
-    isLogedIn = this.userService.isLogged;
+    logoutBox: any;
+    isLogged: boolean = false;
 
-    constructor(public userService: UserService) {
+    constructor(public userService: UserService, private route: Router) {
         userService.loginAnnounced$.subscribe(
-            user => this.updateAccount(user));
+            user => {
+                this.isLogged = this.userService.isLogged;
+                this.updateAccount(user);
+            });
     }
     ngOnInit() { }
 
-    updateAccount(newUser: User){
-        if (newUser) {
+    updateAccount(newUser: User) {
+        if (newUser.id) {
             this.accountName = newUser.name;
+            const options = { /* Reveal Options, if any */ };
+            const $logoutElement = $('.logout-box');
```

Comment:
No sure if this is the best approach - as a rule of thumb you don't need jQuery in a modern SPA as you don't have to do direct changes to the DOM, you only update the state in JS and you let the framework do the changes because it does 10x faster and better than any of us would do.

First of all - do you really need a dropdown for logout? Can't it be a single button?
If yes, try using a Angular directive in HTML template for the dropdown, do not initialize it yourself. (see more here: https://github.com/circlingthesun/angular-foundation-6/tree/master/src/dropdownMenu)
As a final step, you only update the state in JS - _this.showLogout = true_ - and you show that dropdown with respect to that boolean variable.